src/leiningen/ring/war.clj

```diff
@@ -146,6 +146,7 @@
   (let [init-sym    (get-in project [:ring :init])
         destroy-sym (get-in project [:ring :destroy])
         handler-sym (get-in project [:ring :handler])
+        service     (select-keys (:ring project) [:async?])
```

Comment:
Maybe just change this to `async?`, and have the following clause as `(if async? {:async? async?})`